interfaces/cython/cantera/yaml2ck.py

```diff
@@ -0,0 +1,1029 @@
+            line.append(f"{species.name:<16}")
+            line.append(f"{geometry[species.transport.geometry]:>4}")
+            line.append(f"{(species.transport.well_depth / ct.boltzmann):>10.3f}")
+            line.append(f"{(species.transport.diameter * 1e10):>10.3f}")
+            line.append(f"{(species.transport.dipole / DEBYE_CONVERSION):>10.3f}")
+            line.append(f"{(species.transport.polarizability * 1e30):>10.3f}")
+            line.append(f"{species.transport.rotational_relaxation:>10.3f} ! ")
+            if 'note' in species.input_data['transport']:
+                line.append(f"{species.input_data['transport']['note']}")
+
+            text.append("".join(line))
+        
+        return "\n".join(text)
+
+
+def convert(solution, mech_path=None, thermo_path=None, tran_path=None,
+            reorder_reactions=True, overwrite=False):
+    """
+    Writes Cantera solution object to Chemkin-format file(s).
+
+    :param solution:
+        Either the `solution` object being converted or the path leading to it
+    :param mech_path:
+        The Optional path to the output mechanism file, if this is unspecified it will
+        go to the cwd under the name in the `solution` object. 
+        It can be a file name with or without the extension, directory, or a full path
+    :param thermo_path:
+        The Optional path to the output thermo file, if this is unspecified it will
+        go to the cwd under the name in the `solution` object. 
+        It can be a file name with or without the extension, directory, or a full path
+    :param tran_path:
+        The Optional path to the output transport file, if this is unspecified it will
+        go to the cwd under the name in the `solution` object. 
+        It can be a file name with or without the extension, directory, or a full path
+    :param reorder_reactions
+        Boolean flag to reorder reactions based upon molecular weight
+    :param overwrite
+        Boolean flag to allow files to be overwritten or not
+    """
+    if not isinstance(solution, ct.Solution):
+        if not isinstance(solution, PurePath):
+            try:  # try to turn full path into pathlib Path
+                solution = Path(solution)
+            except:
+                raise ValueError(f"Invalid path: {str(solution)}")
+
+            if len(solution.parts) == 1:
+                solution = Path.cwd() / solution
+
+        try:
+            solution = ct.Solution(str(solution))
+        except:
+            raise ValueError(f"Invalid path: {str(solution)}")
+            
+    if mech_path is None:
+        mech_path = Path.cwd() / f"{solution.name}.ck"
+
+    # NOTE: solution.transport_model returns back a string that is None right now
+    if solution.transport_model == "None" or solution.transport_model is None:
+        tran_path = None
+    elif tran_path is None:
+        tran_path = f"{solution.name}.tran"       
+
+    ext = {'mech': 'ck', 'thermo': 'therm', 'tran': 'tran'}
```

Comment:
Please use double quotes for all new code, unless you're formatting a string that has to include double quotes internally.